src/client/app/effects/github.ts

```diff
@@ -141,14 +141,13 @@ export class GitHubEffects {
                 new Snippet.UpdateInfoAction({ id: snippetId, gist: gist.id, gistOwnerId: gist.owner.login })])
         )
         .catch(exception => {
-            if (exception instanceof Error) {
-                exception = JSON.parse(exception.message);
+            let action = null;
+            if (exception instanceof PlaygroundError) {
+                let { snippetId, gistOwnerId } = JSON.parse(exception.message);
```

Comment:
Same as above, check the type to make sure that it's an error that you're supposed to act upon.

Also you should wrap this parsing code with a try/catch.  It would be a shame for the error-handler to throw an error if the input is, for whatever reason, invalid (i.e., it was a PlaygroundError but just with a text message, not the same one as you were throwing above)...